File: sgilt-notifications/src/main/java/net/franzka/sgilt/notifications/notification/dto/NotificationDto.java
package net.franzka.sgilt.notifications.notification.dto;

import java.time.Instant;
import java.util.Map;
import java.util.UUID;

/**
 * Représentation d'une notification exposée à l'API
 * Pas de texte pré-rendu : {@code messageKey} est une clé i18n résolue côté front,
 * {@code params} porte les faits bruts nécessaires à son
 * interpolation (ex. {@code eventTitle}, {@code prestataireName}).
 *
 * @param id         l'identifiant de la notification
 * @param type       le type, en snake_case (ex. {@code "new_request"})
 * @param read       si la notification a été lue
 * @param createdAt  la date de création (sérialisée en ISO-8601)
 * @param messageKey la clé i18n du message affiché
 * @param params     les paramètres d'interpolation du message
 * @param href       la cible de navigation au clic
 */
public record NotificationDto(
        UUID id,
        String type,
        boolean read,
        Instant createdAt,
        String messageKey,
        Map<String, String> params,
        String href
) {}

File: sgilt-front/app/data/notification/dto/NotificationDto.ts
/**
 * DTO — contrat de réponse de l'API GET /notifications
 */

export interface NotificationDto {
  id: string
  type: string
  read: boolean
  createdAt: string
  messageKey: string
  params: Record<string, string>
  href: string
}

/** Sous-ensemble du Page<T> Spring utilisé par le front */
export interface NotificationPageDto {
  items: NotificationDto[]
  hasMore: boolean
}
